### ISSUE TITLE
Question: How can I share textures, meshes and other important assets b/w multiple instances of a GLB loaded from UnityGLTF

### ISSUE THREAD

**Issue by umeraamir-arthur** [KhronosGroup/UnityGLTF#762]
Hi @pfcDorn @hybridherbst 

This one is rather a question. My use case now is to spawn multiple instances of same GLB in a scene. I'm using .Load() from GLTFComponent.cs in multiple game objects. It seems like they're having spikes every time I call .Load() because of the fact that all the assets including textures, meshes are loading multiple times instead of using same assets.

Is there a way to accomplish this situation where they can share important assets instead of creating new ones? Thankyou

**Comment by umeraamir-arthur** [comment]
I've tried this code and I think you're talking about same thing right? This one is insanely good. I can change materials colors and new Instanced Material will be created. Otherwise I don't need to change meshes and textures at runtime. PERFECT for my usecase..... Thanks MAN!!!!

![image](https://github.com/user-attachments/assets/1916656a-3a1f-48aa-9499-4d966005d2bc)
![image](https://github.com/user-attachments/assets/5fab76c4-6a13-4d6f-9f05-6feaa4d73d73)

**Comment by hybridherbst** [comment]
I think you can find all Renderers on the loaded object (before the first instantiate call), get all sharedMaterials, and enable `material.enableInstancing`.

`GLTFSceneImporter` does have an option to enable it as well but I'm not sure if that is exposed for runtime import (cc @pfcDorn)

**Comment by umeraamir-arthur** [comment]
Great! And what about destroying the object....
Is it ??

```
Destroy() == Delete object
DestroyImmediate(object, true) == Delete object with resources
```

Can I replace DestroyImmediate() with something similar to Destroy() ?

**Comment by hybridherbst** [comment]
usually it's
```cs
if (Application.isPlaying)
   Destroy(target);
else
   DestroyImmediate(target);
```

**Comment by umeraamir-arthur** [comment]
I found this utility on LastLoadedScene. This MonoBehaviour is attached to the top most scene and is doing everything I needed without using custom Instantiate() calls and I can do Destroy() calls here.
![image](https://github.com/user-attachments/assets/51a86ef0-9554-4682-8f67-305a50ec2626)

**Comment by umeraamir-arthur** [comment]
One thing I noticed that using this above method, there is a memory leak of animation clips as shown in the code below. I suppose that GLTFComponent.cs has reference of these AnimationClips that might be the reason. Even DestroyImmediate(object, true) was also not able to delete those clips and memory profilers shows them as leaked objects.

If I use Destroy(gameObject) which is the current game object with GLTFComponent.cs script then it was able to delete everything including those clips.

![image](https://github.com/user-attachments/assets/6ab0a189-f5eb-4e67-a046-19e78e80bbb5)

**Comment by hybridherbst** [comment]
cc @pfcDorn can you look at the leak so that we can close this issue?

**Comment by pfcDorn** [comment]
The Animation Clip issue will be fixed with 48355bb91649026dbd5c91f15908051edbb0d14a :)
Thanks for testing!
